### PR TITLE
Add query, langType, status, error in AsyncQueryJobMetadata

### PR DESCRIPTION
**Pull request by ykmr1224**
### Description
- Add query, langType, status, error in AsyncQueryJobMetadata
 - This is part of https://github.com/opensearch-project/opensearch-spark/issues/602. As the first change, add attributes to AsyncQueryJobMetadata, and populate those fields. It won't save to index yet (that change would be separate PR).

### Related Issues
https://github.com/opensearch-project/opensearch-spark/issues/602

### Check List
- [x] New functionality includes testing.
- [ ] New functionality has been documented.
 - [x] New functionality has javadoc added.
 - [ ] New functionality has a user manual doc added.
- [ ] API changes companion pull request [created](https://github.com/opensearch-project/opensearch-api-specification/blob/main/DEVELOPER_GUIDE.md).
- [x] Commits are signed per the DCO using `--signoff`.
- [ ] Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/sql/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: async-query-core/src/main/java/org/opensearch/sql/spark/asyncquery/model/QueryState.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+package org.opensearch.sql.spark.asyncquery.model;
+
+import java.util.Arrays;
+import java.util.Locale;
+import java.util.Map;
+import java.util.stream.Collectors;
+import lombok.Getter;
+
+@Getter
+public enum QueryState {
```
Comment (vamsi-amazon) [inline async-query-core/src/main/java/org/opensearch/sql/spark/asyncquery/model/QueryState.java:15]: How did we come up with these? Do these also cover emr job status. For eg: `scheduled` when emr is yet to pick up the job.

**Review by vamsi-amazon — COMMENTED** [review]
LGTM

**Review by vamsi-amazon — DISMISSED** [review]

File: async-query-core/src/main/java/org/opensearch/sql/spark/asyncquery/model/QueryState.java (hunk 1 above)
Comment (ykmr1224) [reply async-query-core/src/main/java/org/opensearch/sql/spark/asyncquery/model/QueryState.java:15]: It is coming from StatementState. I think `scheduled` you mentioned is equivalent to `waiting`. Is that more like EMR job status?

**Review by seankao-az — DISMISSED** [review]

**Review by seankao-az — APPROVED** [review]

**Review by vamsi-amazon — DISMISSED** [review]

**Review by seankao-az — DISMISSED** [review]

**Review by seankao-az — APPROVED** [review]

File: async-query-core/src/main/java/org/opensearch/sql/spark/asyncquery/model/QueryState.java (hunk 1 above)
Comment (vamsi-amazon) [reply async-query-core/src/main/java/org/opensearch/sql/spark/asyncquery/model/QueryState.java:15]: yeah.

**Review by vamsi-amazon — APPROVED** [review]

**Pull request merged**